podpac/core/coordinates/coordinates.py

```diff
@@ -1790,28 +1738,37 @@ def full_unstacked_resolution(dim):
             1 / self.CRS.ellipsoid.inverse_flattening,
         )
 
+        # validate dims:
+
         # main execution loop
         for dim in self.dims:
             # Is the dim lat/lon?
             if not check_horizontal(dim):
                 continue
+            # Put this check inside the loop to avoid checking if stacked coords
+            if dim == "lat":
+                if "lon" not in self.dims:
+                    return ValueError("Need both lat and lon for resolution.")
+            if dim == "lon":
+                if "lat" not in self.dims:
+                    return ValueError("Need both lat and lon for resolution.")

```

Comment:
Hmmm, I would just check `if lat not in self.udims or lon not in self.udims` once outside of the loop.

What if the dims are `['lat_time']`? That case should fail. Or `['lat_time', 'lon']` which should work? I'm not sure your loop and functions handle the `['lat_time', 'lon']` case at all. I know these cases seem weird, maybe you guys can decide not to handle them. @mpu-creare are these sorts of coordinates still allowed? @CFoye-Creare Am I right that this case isn't handled?

Something like this would be more general:

```
def horizontal_resolution(self, units, restype):
    if 'lat' not in self.udims or 'lon' not in self.udims:
        raise ValueError("horizontal_resolution requires lat and lon dimensions.")
    d = {}
    for dim, c in self.items():
        if 'lat' in c.dims and 'lon' in c.dims:
            d[dim] = stacked_horizontal_resolution(self['lat'], self['lon'], units, restype)
        elif 'lat' in c.dims:
            d[dim] = unstacked_lat_horiziontal_resolution(self['lat'], self['lon'], units, restype)
        elif 'lon' in c.dims:
            d[dim] = unstacked_lon_horiziontal_resolution(self['lat'], self['lon'], units, restype)
    return d
```

I still prefer a version that moves those functions to the Coordinates1d and StackedCoordinates classes, but regardless, at least passing the coordinates in as an argument is better in principle because it makes the behavior explicit rather than implicit. And at least moving the helper functions out of horizontal_resolution would be better because then you can't use the other dimention in the "wrong scope", plus you can unit test them. In general, the inline functions lead you to make assumptions that aren't true and makes readability and testing harder.

Reply:
I agree. We talked about it this morning and are going to go with the 1dCoordinates and StackedCoordinates implementation.

Reply:
I'm not sure I understand how the `'lat_time'` case would work. It was my understanding that we only on horizontal resolution.

Reply:
Got it.